forc-plugins/forc-doc/src/main.rs

```diff
@@ -32,24 +28,31 @@ pub(crate) const ASSETS_DIR_NAME: &str = "static.files";
 
 /// Information passed to the render phase to get TypeInfo, CallPath or visibility for type anchors.
 #[derive(Clone)]
-struct RenderPlan {
+struct RenderPlan<'te, 'de> {
```

Comment:
```suggestion
struct RenderPlan<'a> {
    document_private_items: bool,
    type_engine: &'a TypeEngine,
    decl_engine: &'a DeclEngine,
}
impl<'a> RenderPlan<'a> {
    fn new(
        document_private_items: bool,
        type_engine: &'a TypeEngine,
        decl_engine: &'a DeclEngine,
    ) -> RenderPlan<'a> {
        Self {
            document_private_items,
            type_engine,
            decl_engine,
        }
    }
}
```

We can probably share the same lifetime between the type and decl engines here.

Reply:
> Can we add a test to verify the fix?

Hm, I'm not sure how you would test this since the fix doesn't really change the output but instead changes the strategy of building to use `forc-pkg`'s `BuildPlan`.